### DISCUSSION TITLE
[Feature] Support skip TLS host verify on ElasticSearch

### DISCUSSION THREAD

**Discussion by chanjarster** [apache/skywalking#8154] [Q&A]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

Please add support to config skip TLS server auth when connect to ElasticSearch storage.

### Use case

Deploying a  SkyWalking instance by Helm Chart with external ElasticSearch storage (https enabled) is inconvenient. 

You have to config a truststore and it's hard in k8s.

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by wu-sheng** [comment]
Skipping TLS is supported through system environment setting. This is not a feature actually.

**Reply by chanjarster** [reply]
Would you like to tell me which system env is for TLS skipping?

**Reply by wu-sheng** [reply]
https://skywalking.apache.org/docs/main/v8.8.1/en/setup/backend/backend-storage/#elasticsearch

The offical document is recomended as always.

**Reply by chanjarster** [reply]
I read the doc, and no env found for skipping TLS verify.

I read the code [ElasticSearchBuilder.java][1], there is no snippet about skip TLS verify.

[1]: https://github.com/apache/skywalking/blob/a2897db4715092b8e4079fd57b462606deee2fdf/oap-server/server-library/library-elasticsearch-client/src/main/java/org/apache/skywalking/library/elasticsearch/ElasticSearchBuilder.java#L152-L163

**Reply by chanjarster** [reply]
@wu-sheng I did some dig, really didn't find any TLS verify skip options or logic in `ElasticSearchBuilder`, `ElasticSearchClient`, `StorageModuleElasticsearchConfig`, `StorageModuleElasticsearchProvider`.

And I notice that `com.linecorp.armeria.client.ClientFactoryBuilder`, which is used by `ElasticSearchBuilder`,  has a property `private boolean tlsNoVerifySet` (default is false), and it's not modified in SkyWalking code.

So I plan to give a PR on this issue.

Any suggestions?

**Reply by wu-sheng** [reply]
I am still not sure what do you mean skipping. 
What I can see in the `StorageModuleElasticsearchProvider`, is once `protocol=http` and `trustStorePath` and `trustStorePass` are empty, there is nothing relating to TLS. 
Generally I am not opening TLS on local testing every time.

**Reply by wu-sheng** [reply]
> So I plan to give a PR on this issue.
Any suggestions?

I am not sure about this requirement, so, hard to tell the suggestion for now.

**Reply by chanjarster** [reply]
Well... I'll create a PR anyway, on branch 8.8.x.

**Reply by wu-sheng** [reply]
That branch wouldn't accept a pull request, unless it is a CVE fix or critical issue.

**Reply by wu-sheng** [reply]
If you want to propose anything, target to master branch please.

**Reply by chanjarster** [reply]
OK

**Reply by tirelibirefe** [reply]
> Skipping TLS is supported through system environment setting. This is not a feature actually.

Hello,
Could you please send a reference link? I need to this...

**Reply by wu-sheng** [reply]
Both PRs are rejected because we don't plan to open this exception as we can't see real use cases.
- https://github.com/apache/skywalking/pull/8179
- https://github.com/apache/skywalking/pull/8215

**Comment by tirelibirefe** [comment]
does it mean that it's not possible to use ElasticSearch as backend if ES is configured with self-signed certificate?

**Reply by wu-sheng** [reply]
Why it isn't possible? Those discussions in the PRs have provided full context and recommended solutions. But none of them says `self-signed` is an issue from the beginning.

**Reply by tirelibirefe** [reply]
I use self-signed tls with ES and I would like to use that ES as backend of Skywalking.
If this config is supported by Skywalking, Where are environment variables to skip ssl check for ES connection? 
Could you please advise?
Thanks

**Reply by tirelibirefe** [reply]
is there a plan or something in road-map to support ElasticSearch with self-signed certificates?

**Reply by chanjarster** [reply]
@tirelibirefe self-signed tls cert is supported, all you need to do are:

1. make a JKS file, which contains your self-signed cert or private CA
2. put that JKS file somewhere on SkyWalking server 
3. configure `storage.elasticsearch.trustStorePath` point to that JKS file
4. configure `storage.elasticsearch.trustStorePass` if JKS file has a passphrase

ref link:

* https://skywalking.apache.org/docs/main/v8.9.1/en/setup/backend/configuration-vocabulary/
